src/shared/telemetry/defaultTelemetryService.ts

```diff
@@ -279,11 +292,18 @@ export class DefaultTelemetryService implements TelemetryService {
 
     /**
      * Verifies that telemetry queue only has passive metrics (except for some known
-     * special-cases).
+     * special-cases). Logs error or throws depending on if this is a prod version or not (respectively)
+     *
+     * Should only be called once per session.
      *
-     * Used while validating extension activation.
+     * Specifically designed to be used while validating extension activation.
      */
-    public assertPassiveTelemetry(didReload: boolean) {
+    public assertOnlyPassiveTelemetryInQueue(didReload: boolean) {
```

Comment:
"only" is mostly redundant, so not worth the extra verbosity. This rationale can be applied generally to all naming choices.

Function names should be meaningful enough, but don't need to describe their total spec. The docstring is where ambiguity can be resolved. And because "assertPassiveTelemetry" is already a very unique name, there's already near-zero risk of ambiguity. Names live in the context of their codebase.